### DISCUSSION TITLE
Multiple policies for a queue

### DISCUSSION THREAD

**Discussion by lechen26** [rabbitmq/rabbitmq-server#2819] [Other]
Hi

i'm using rabbitmq 3.8.2 and recently noticed only single policy get applied to queue.

i have a policy that sets ha-all on all queues and another one that deleted queue (expiration) after specific time of not used, except specific queues which i wanted to exclude.
seems like only the first policy get applied and i see now its by design.

was there any improvements here?
create a single policy with both things will cause the excluded queues to not have also the first policy so its not somethiing we want.

any idea?
10x

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
There were no changes to policies in several years, except for [Operator policies](https://www.rabbitmq.com/parameters.html#operator-policies) which address the most pressing need of RabbitMQ-as-a-Service deployments.

There are no plans to change how policies or policy priorities work.

While not really related, some members of our team would strongly question a combination of TTL and queue mirroring (or using quorum queues with TTL). Is your data transient and can be deleted or so important everything has to be replicated? Those are conflicting choices and you have to decide. Mirroring everything is almost always unnecessary.

**Reply by lechen26** [reply]
thanks @michaelklishin.
interesting.

we use heavily rabbit as the message bus for all our services and we indeed use to mirror all queues.
all messages on our queues supposed to be consumed by our services and all of them are persistent (queues are durable)

you still think mirroring is unneceessary?
we assume its a most if we want to have some kind of HA on rabbit in case of one rabbit node evicted (GKE) for any reason.
thanks

**Reply by jxsl13** [reply]
he said that mirroring in combination with time to live policy settings is contradicting.
If messages need to be consumed, they do not have a TTL setting.

**Reply by lechen26** [reply]
thanks. i did not set TTL for the  messages.
we wanted to have expiration on the queues so if  60 days no one worked with them  (idle) we delete them.
